text/0000-more-iter-methods.md

```diff
@@ -0,0 +1,95 @@
+- Feature Name: more-iter-methods
+- Start Date: 2016-10-16
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the addition of various methods to the Iter class of the itertools package. The methods included are based on functions from [Rust's Iterator](https://doc.rust-lang.org/std/iter/trait.Iterator.html) and [Elixir's Enum](http://elixir-lang.org/docs/stable/elixir/Enum.html).
+
+# Motivation
+
+The purpose of these additional methods is to increase the functionality of the Iter class.
+
+# Detailed design
+
+The following methods will be added to the Iter class:
+```pony
+fun ref all(f: {(A!): Bool ?} box): Bool =>
+    """
+    Return false if at least one value of the iterator fails to match the
+    predicate `f`, otherwise return true.
+    """
+
+fun ref any(f: {(A!): Bool ?} box): Bool =>
+    """
+    Return true if at least one value of the iterator matches the predicate `f`,
+    otherwise return false.
```

Comment:
Is it short-circuiting?  (Do all of elements get the function executed, or does execution stop when a `true` is reached?)  This should be made clear in the doc string.